auth/src/main/java/software/amazon/awssdk/auth/signer/AbstractAws4Signer.java

```diff
@@ -378,11 +379,9 @@ private void addDateHeader(SdkHttpFullRequest.Builder mutableRequest, String dat
      * Generates an expiration time for the presigned url. If user has specified
      * an expiration time, check if it is in the given limit.
      */
-    private long generateExpirationTime(Instant expirationTime) {
-
-        long expirationInSeconds = expirationTime != null
-                                   ? expirationTime.getEpochSecond()
-                                   : SignerConstant.PRESIGN_URL_MAX_EXPIRATION_SECONDS;
+    private long generateExpirationTime(Optional<Instant> expirationTime) {
```

Comment:
Can we wrap it in optional internally instead of taking optional as a param?

Reply:
Fixed. Updated the parameter to be `signerParams` rather than optional